models/dataset.go

```diff
@@ -507,32 +498,15 @@ func ValidateDataset(ctx context.Context, dataset *Dataset) error {
 
 }
 
-func santisedGeneralDetails(input []GeneralDetails) ([]GeneralDetails, []string) {
-	sanitisedDetails := make([]GeneralDetails, 0)
-	invalidFields := make([]string, 0)
-
-	for _, original := range input {
-		cleanHref := strings.TrimSpace(original.HRef)
-		_, err := url.Parse(cleanHref)
+func validateGeneralDetails(generalDetails []GeneralDetails, identifier string) (invalidFields []string) {
+	for i, gd := range generalDetails {
+		_, err := url.Parse(gd.HRef)
 		if err != nil {
-			invalidFields = append(invalidFields, "Href")
+			invalidFields = append(invalidFields, fmt.Sprintf("%s[%d].HRef", identifier, i))
 			continue
 		}
-
-		cleanDetails := GeneralDetails{
-			Description: original.Description,
-			HRef:        cleanHref,
-			Title:       original.Title,
-		}
-
-		sanitisedDetails = append(sanitisedDetails, cleanDetails)
-	}
-
-	if len(invalidFields) > 0 {
-		return input, invalidFields
 	}
-
-	return sanitisedDetails, nil
+	return
 }
 
 func trimHref(ctx context.Context, generalDetails []GeneralDetails) error {
```

Comment:
The `trimHref()` function is now unused and you can remove it.

Reply:
I see it has been removed.